packages/contact-center/task/src/helper.ts

```diff
@@ -0,0 +1,142 @@
+import {useState, useEffect, useCallback, useRef} from 'react';
+import {TASK_EVENTS, UseTaskListProps, UseTaskProps} from './task.types';
+import {ITask} from '@webex/plugin-cc';
+
+// Hook for managing the task list
+export const useTaskList = (props: UseTaskListProps) => {
+  const {cc} = props;
+  const [taskList, setTaskList] = useState<ITask[]>([]);
+  const isEventRegistered = useRef(false); // Ensure the event is registered only once
```

Comment:
what is this ref for??

Reply:
we are using this as a boolean we dont need this as a ref. ref is used when we assign it to an html element

Reply:
removed